src/foam/core/XML.js

```diff
@@ -0,0 +1,465 @@
+/**
```

Comment:
This isn't copyright Google if you wrote it. Use this header instead:

/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

And add yourself to the foam2/AUTHORS file.